### ISSUE TITLE
argsort这个OP是否反向传播有问题

### ISSUE THREAD

**Issue by huangguanMayday** [PaddlePaddle/Paddle#24216]
logits是batch*batch的tensor:

将
hardest_neg_score = fluid.layers.reduce_max(logits, dim=1, keep_dim=True) 

换成：
hardest_neg_score3 = fluid.layers.argsort(input=logits, axis=1, name='hardest_neg_score3') 
hardest_neg_score3 = hardest_neg_score3[0]
hardest_neg_score = fluid.layers.slice(input=hardest_neg_score3, axes=[1], starts=[int(batch_size - 1)], ends=[int(batch_size)])

模型收敛从没问题变成有问题。

验证过，两者的前向传播的结果是一致的。

**Comment by yiicy** [comment]
你好，我们已经找相关的同学排查该问题

**Comment by wawltor** [comment]
> logits是batch*batch的tensor:
> 
> 将
> hardest_neg_score = fluid.layers.reduce_max(logits, dim=1, keep_dim=True)
> 
> 换成：
> hardest_neg_score3 = fluid.layers.argsort(input=logits, axis=1, name='hardest_neg_score3')
> hardest_neg_score3 = hardest_neg_score3[0]
> hardest_neg_score = fluid.layers.slice(input=hardest_neg_score3, axes=[1], starts=[int(batch_size - 1)], ends=[int(batch_size)])
> 
> 模型收敛从没问题变成有问题。
> 
> 验证过，两者的前向传播的结果是一致的。

argsort op的输出结果本身是stop_gradient的，因为属于tensor 生成op；
建议以下修改
hardest_neg_score3 = fluid.layers.argsort(input=logits, axis=1, name='hardest_neg_score3')
hardest_neg_score3 = hardest_neg_score3[0]
hardest_neg_score3.stop_gradient = False
hardest_neg_score = fluid.layers.slice(input=hardest_neg_score3, axes=[1], starts=[int(batch_size - 1)], ends=[int(batch_size)])
